modules/installer.nix

```diff
@@ -35,6 +39,132 @@ with builtins; with lib; {
         exec ${builtins.unsafeDiscardStringContext config.users.users.root.shell} "$@"
       '';
 
+      updater =
+        let
+          stage2 = pkgs.writeScript "updater-stage-2.sh" ''
+            #!${busybox}/bin/sh
+            set -euo pipefail
+
+            # Set PATH to include busybox
+            export PATH=${busybox}/bin:$PATH
+
+            # Root check
+            if [ `id -u` != "0" ]; then
+              echo "This script must be run as root" >&2
+              exit 1
+            fi
+
+
+            function prompt_yn {
+              while true; do
+                read -p "$@ [y/n] " result
+                case $result in
+                  [Yy]* ) return 0;;
+                  [Nn]* ) return 1;;
+                  * ) echo "Please answer yes or no.";;
+                esac
+              done
+            }
+
+            #Parse arguments
+            GC=true
+            REBUILD=false
+            for arg in "$@"; do
+              case $arg in
+                --no-gc)
+                  GC=false
+                  ;;
+                --rebuild)
+                  REBUILD=true
+                  ;;
+                *)
+                  echo "Unknown argument: $arg"
+                  exit 1
+                  ;;
+              esac
+            done
+
+
+            # Check if /etc/nixos/nixos-wsl exists
+            if ! [ -d /etc/nixos/nixos-wsl ]; then
+              echo "/etc/nixos/nixos-wsl was not found! Can not update this system" >&2
+              exit 1
+            fi
+
+            # Unpack rootfs
+            echo "Unpacking..."
+            mkdir -p /tmp/nixos-wsl-updater/rootfs
+            ${pv}/bin/pv ${rootfs} | tar -C /tmp/nixos-wsl-updater/rootfs -xz
+
+            # Detect flake.nix
+            FLAKE=false
+            if [ -f /etc/nixos/flake.nix ]; then
+              echo "Flake detected!"
+              echo "Please update your nixpkgs input to \"github:nixos/nixpkgs/nixos-${config.system.nixos.release}\" and run nixos-rebuild to complete the update"
+              echo "Press enter to continue..."
+              read
+              FLAKE=true
+            fi
+
+            # Check that nixos-rebuild builds the current configuration
+            EXTERNAL=false
+            if ! $FLAKE; then
```

Comment:
Why do we check this? Even if this fails I don't think it should abort the installation.

Reply:
It checks that `nixos-rebuild` will build the config you are currently running, and if it doesn't, it skips updating the nixpkgs channel and the final `nixos-rebuild` steps and tells you to perform it manually. The install does not get cancelled